File: plugins/OrchidJavadoc/src/main/java/com/eden/orchid/javadoc/pages/JavadocClassPage.java
package com.eden.orchid.javadoc.pages;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.options.annotations.Option;
import com.eden.orchid.api.theme.pages.OrchidPage;
import com.eden.orchid.javadoc.JavadocModel;
import com.eden.orchid.javadoc.resources.ClassDocResource;
import com.sun.javadoc.ClassDoc;

public class JavadocClassPage extends OrchidPage {

    private JavadocPackagePage packagePage;
    private ClassDoc classDoc;

    @Option
    public String deprecated;

    public JavadocClassPage(OrchidContext context, ClassDoc classDoc) {
        super(new ClassDocResource(context, classDoc), "javadocClass", classDoc.typeName());
        this.classDoc = classDoc;
    }

    public String getClassType() {
        if(this.classDoc.isInterface()) {
            return "interface";
        }
        else if(this.classDoc.isAnnotationType()) {
            return "@interface";
        }
        else if(this.classDoc.isEnum()) {
            return "enum";
        }

        return "class";
    }

    public JavadocPackagePage getPackagePage() {
        return packagePage;
    }

    public ClassDoc getClassDoc() {
        return classDoc;
    }

    public void setPackagePage(JavadocPackagePage packagePage) {
        this.packagePage = packagePage;
    }

    public void setClassDoc(ClassDoc classDoc) {
        this.classDoc = classDoc;
    }

    public OrchidPage getClassPage(String className) {
        JavadocModel javadocModel = context.getInjector().getInstance(JavadocModel.class);

        for(JavadocClassPage classPage : javadocModel.getAllClasses()) {
            if(classPage.classDoc.qualifiedName().equals(className)) {
                return classPage;
            }
            else if(classPage.classDoc.name().equals(className)) {
                return classPage;
            }
        }

        return null;
    }
}

File: plugins/OrchidJavadoc/src/main/kotlin/com/eden/orchid/javadoc/pages/JavadocClassPage.kt
package com.eden.orchid.javadoc.pages

import com.copperleaf.javadoc.json.models.JavaClassDoc
import com.eden.orchid.api.OrchidContext
import com.eden.orchid.api.options.annotations.Archetype
import com.eden.orchid.api.options.annotations.Description
import com.eden.orchid.api.options.archetypes.ConfigArchetype
import com.eden.orchid.javadoc.JavadocGenerator
import com.eden.orchid.javadoc.models.JavadocModel
import com.eden.orchid.javadoc.resources.ClassDocResource

@Archetype(value = ConfigArchetype::class, key = "${JavadocGenerator.GENERATOR_KEY}.classPages")
@Description(value = "Documentation for a Java class.", name = "Java class")
class JavadocClassPage(
        context: OrchidContext,
        val classDoc: JavaClassDoc
) : BaseJavadocPage(ClassDocResource(context, classDoc), "javadocClass", classDoc.name) {

    lateinit var model: JavadocModel
    var packagePage: JavadocPackagePage? = null

    override val itemIds: List<String>
        get() = listOf(
            classDoc.qualifiedName,
            classDoc.name
        )
}
